### PR TITLE
Adds option to return profiles as dict

### PR DESCRIPTION
**Pull request by purujitgoyal**

### CODE REVIEW

File: openreview/tools.py [hunk 1]
```diff
@@ -134,18 +137,29 @@ def get_profiles(client, ids_or_emails, with_publications=False):
...
+    profiles_as_dict = {}
 
     batch_size = 100
     for i in range(0, len(ids), batch_size):
         batch_ids = ids[i:i+batch_size]
         batch_profiles = client.search_profiles(ids=batch_ids)
         profiles.extend(batch_profiles)
 
+    if as_dict:
+        for i, profile in enumerate(profiles):
```
Comment (melisabok) [inline openreview/tools.py:149]: is this assuming the profiles are returned in the same order? also what happens with the ids that don't return any profile (it should happen). 

I would build the dictionary using the `ids` arrays instead, same as you are doing with the emails. You would need to find the profile that contains that `id` into the `profile.content['names']` array. Maybe using a dictionary where the key are all the alternate name can help.

File: openreview/tools.py [hunk 2]
```diff
@@ -146,8 +146,14 @@ def get_profiles(client, ids_or_emails, with_publications=False, as_dict=False):
         profiles.extend(batch_profiles)
 
     if as_dict:
-        for i, profile in enumerate(profiles):
-            profiles_as_dict[ids[i]] = profile
+        for profile in profiles:
+            for name in profile.content.get("names", {}):
```
Comment (melisabok) [inline openreview/tools.py:150]: names is an array so the default value should be [].

File: openreview/tools.py [hunk 3]
```diff
@@ -146,8 +146,14 @@ def get_profiles(client, ids_or_emails, with_publications=False, as_dict=False):
...
+                if name.get("username") in ids:
+                    profiles_as_dict[name.get("username")] = profile
+                    continue
```
Comment (melisabok) [inline openreview/tools.py:153]: If we pass two different tilde ids that belongs to the same profile we are going to get a profile for one tilde id and None for the other. I think you can reproduce this adding more tests. I believe there are profiles with alternate names. 

You must check that all the `ids` have a profile. I would suggest to do the following:

```
profiles_by_name = {}
for profile in profiles:
      for name in profile.content.get('names', []):
            if name.get('username'):              
                  profiles_by_name[name.get('username')] = profile

for id in ids:
      profiles_as_dict[id] = profile_by_name.get(id)
```

Comment (purujitgoyal) [reply openreview/tools.py:153]: Emails will have same issue

Comment (melisabok) [reply openreview/tools.py:153]: could you provide a test? that is the best way to verify if it fails.

File: openreview/tools.py [hunk 4]
```diff
@@ -3,7 +3,9 @@
 from __future__ import absolute_import, division, print_function, unicode_literals
 
 import json
+import logging
 import os
+from collections import OrderedDict
```
Comment (melisabok) [inline openreview/tools.py:8]: do you need this import?

File: openreview/tools.py [hunk 5]
```diff
@@ -3,7 +3,9 @@
 from __future__ import absolute_import, division, print_function, unicode_literals
 
 import json
+import logging
```
Comment (melisabok) [inline openreview/tools.py:6]: do you need this import?

File: openreview/tools.py [hunk 6]
```diff
@@ -152,18 +155,37 @@ def get_profiles(client, ids_or_emails, with_publications=False):
...
     for i in range(0, len(ids), batch_size):
         batch_ids = ids[i:i+batch_size]
         batch_profiles = client.search_profiles(ids=batch_ids)
         profiles.extend(batch_profiles)
 
+    if as_dict:
+        profiles_by_name = {}
+        for profile in profiles:
+            for name in profile.content.get("names", []):
+                profiles_by_name[name.get("username")] = profile
```
Comment (melisabok) [inline openreview/tools.py:170]: name.get("username") can return None here, is it ok to have a None key?

Comment (purujitgoyal) [reply openreview/tools.py:170]: Yeah, it won't cause any issues.

File: openreview/tools.py (hunk 4 above)
Comment (purujitgoyal) [reply openreview/tools.py]: Removed.

File: openreview/tools.py (hunk 5 above)
Comment (purujitgoyal) [reply openreview/tools.py]: Removed.

**Review by melisabok — APPROVED** [review]

**Pull request merged**